task/backend/coordinator: log from a goroutine after a test causes a failure

When running the `task/backend/coordinator` tests, this message will sometimes appear.

```
panic: Log in goroutine after TestCoordinator_ForceRun has completed
goroutine 23 [running]:
testing.(*common).logDepth(0xc000156400, 0xc0004f2360, 0x56, 0x3)
        /usr/local/Cellar/go/1.12.1/libexec/src/testing/testing.go:634 +0x3a6
testing.(*common).log(...)
        /usr/local/Cellar/go/1.12.1/libexec/src/testing/testing.go:614
testing.(*common).Logf(0xc000156400, 0x1b768c6, 0x2, 0xc0006f0100, 0x1, 0x1)
        /usr/local/Cellar/go/1.12.1/libexec/src/testing/testing.go:649 +0x7f
go.uber.org/zap/zaptest.testingWriter.Write(0x1e7a620, 0xc000156400, 0x0, 0xc00028bc00, 0x57, 0x400, 0xc00028bc00, 0x19e02c0, 0xc0006f00f0)
        /Users/jsternberg/go/pkg/mod/go.uber.org/zap@v1.9.1/zaptest/logger.go:114 +0xfc
go.uber.org/zap/zapcore.(*ioCore).Write(0xc0003ebc80, 0x2, 0xbf23523fcf513038, 0x754d9e32, 0x27d8d80, 0x0, 0x0, 0x1b8573f, 0x11, 0x0, ...)
        /Users/jsternberg/go/pkg/mod/go.uber.org/zap@v1.9.1/zapcore/core.go:90 +0x107
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0002024d0, 0xc000730240, 0x1, 0x1)
        /Users/jsternberg/go/pkg/mod/go.uber.org/zap@v1.9.1/zapcore/entry.go:215 +0x119
go.uber.org/zap.(*Logger).Error(0xc000518780, 0x1b8573f, 0x11, 0xc000730240, 0x1, 0x1)
        /Users/jsternberg/go/pkg/mod/go.uber.org/zap@v1.9.1/logger.go:203 +0x7f
github.com/influxdata/influxdb/task/backend/coordinator.(*Coordinator).claimExistingTasks(0xc0003ebcb0)
        /Users/jsternberg/go/src/github.com/influxdata/influxdb/task/backend/coordinator/coordinator.go:63 +0x2ee
created by github.com/influxdata/influxdb/task/backend/coordinator.New
        /Users/jsternberg/go/src/github.com/influxdata/influxdb/task/backend/coordinator/coordinator.go:41 +0xe2
```

This is likely caused by logging in a goroutine after the test has already completed. The test should wait for all spawned goroutines to stop before finishing.

--- a/task/backend/coordinator/coordinator.go
+++ b/task/backend/coordinator/coordinator.go
@@ -15,7 +15,8 @@ type Coordinator struct {
 	logger *zap.Logger
 	sch    backend.Scheduler
 
-	limit int
+	limit         int
+	claimExisting bool
 }
 
 type Option func(*Coordinator)
@@ -26,19 +27,29 @@ func WithLimit(i int) Option {
 	}
 }
 
+// WithoutExistingTasks allows us to skip claiming tasks already in the system.
+func WithoutExistingTasks() Option {
+	return func(c *Coordinator) {
+		c.claimExisting = false
+	}
+}
+
 func New(logger *zap.Logger, scheduler backend.Scheduler, ts platform.TaskService, opts ...Option) *Coordinator {
 	c := &Coordinator{
-		logger:      logger,
-		sch:         scheduler,
-		TaskService: ts,
-		limit:       1000,
+		logger:        logger,
+		sch:           scheduler,
+		TaskService:   ts,
+		limit:         1000,
+		claimExisting: true,
 	}
 
 	for _, opt := range opts {
 		opt(c)
 	}
 
-	go c.claimExistingTasks()
+	if c.claimExisting {
+		go c.claimExistingTasks()
+	}
 
 	return c
 }

--- a/task/backend/coordinator/coordinator_test.go
+++ b/task/backend/coordinator/coordinator_test.go
@@ -109,7 +109,7 @@ func TestCoordinator(t *testing.T) {
 	ts := inmemTaskService()
 	sched := mock.NewScheduler()
 
-	coord := coordinator.New(zaptest.NewLogger(t), sched, ts)
+	coord := coordinator.New(zaptest.NewLogger(t), sched, ts, coordinator.WithoutExistingTasks())
 	createChan := sched.TaskCreateChan()
 	releaseChan := sched.TaskReleaseChan()
 	updateChan := sched.TaskUpdateChan()
@@ -211,7 +211,7 @@ func TestCoordinator_DeleteUnclaimedTask(t *testing.T) {
 	ts := inmemTaskService()
 	sched := mock.NewScheduler()
 
-	coord := coordinator.New(zaptest.NewLogger(t), sched, ts)
+	coord := coordinator.New(zaptest.NewLogger(t), sched, ts, coordinator.WithoutExistingTasks())
 
 	// Create an isolated task directly through the store so the coordinator doesn't know about it.
 	task, err := ts.CreateTask(context.Background(), platform.TaskCreate{OrganizationID: 1, Token: "token", Flux: script})
@@ -281,7 +281,7 @@ func TestCoordinator_ForceRun(t *testing.T) {
 	ts := inmemTaskService()
 	sched := mock.NewScheduler()
 
-	coord := coordinator.New(zaptest.NewLogger(t), sched, ts)
+	coord := coordinator.New(zaptest.NewLogger(t), sched, ts, coordinator.WithoutExistingTasks())
 
 	// Create an isolated task directly through the store so the coordinator doesn't know about it.
 	task, err := coord.CreateTask(context.Background(), platform.TaskCreate{OrganizationID: 1, Token: "token", Flux: script})

